### PR TITLE
Feature/gh 735 payment tokens vc

### LINKED ISSUE #735: PaymentMethod View Controller
- TableVC
- Header View
- Basic cell adjustment

### PR DESCRIPTION
**Pull request by sche**
closes #735

### CODE REVIEW

File: SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        addGestureRecognizer(tapRecognizer)
+        isUserInteractionEnabled = true
+    }
+
+    @objc private func didTap() {
+        delegate?.didTap()
+    }
+
+    public func setInfoText(_ text: String) {
+        let attributedString = NSMutableAttributedString(string: "\(text)\u{00A0}\(infoSuffix)")
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift:30]: would be nice to have explanation why this thing is the way it is

File: SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//  Copyright © 2019 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+public protocol InfoLabelDelegate: class {
+    func didTap()
+}
+
+public class InfoLabel: UILabel {
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift]: can inherit from BaseCustomLabel

File: SafeUIKit/SafeAppUI/ReusableDetachedControllers/PaymentMethod/PaymentMethodHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        super.awakeFromNib()
+        backgroundView = UIView()
+        backgroundView!.backgroundColor = .white
+        headingView.backgroundColor = ColorName.paleGrey.color
+        descriptionInfoLabel.setInfoText(Strings.description)
+        descriptionInfoLabel.delegate = self
+        let headingAttributes: [NSAttributedString.Key: Any] =
+            [.font: UIFont.systemFont(ofSize: 10, weight: .bold),
+             .foregroundColor: ColorName.lightGreyBlue.color,
+             .kern: 2]
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeAppUI/ReusableDetachedControllers/PaymentMethod/PaymentMethodHeaderView.swift]: looks like these attributes are repeated here and in `BackgroundHeaderFooterView.swift`. I'd suggest making a new style inheriting from `AttributedStringStyle`. For examples, look for `ContentHeaderStyle` in the project.

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 4]
```diff
@@ -404,6 +404,18 @@ public class WalletApplicationService: Assertable {
...
 
+    /// Changes payment token for the selected wallet.
+    /// New payment token is whitelisted to be displayed on Assets screen.
+    ///
+    /// - Parameter token: token data.
+    public func changePaymentToken(_ token: TokenData) {
+        let wallet = selectedWallet!
+        wallet.changeFeePaymentToken(Address(token.address))
+        DomainRegistry.walletRepository.save(wallet)
+        guard !token.isEther else { return }
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift]: Just have a feeling that the `if` would be simpler:

```
if token.isEther { return }
```

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift [hunk 5]
```diff
@@ -394,8 +394,8 @@ public class WalletApplicationService: Assertable {
     /// - Returns: token data array.
     public func paymentTokens() -> [TokenData] {
         guard let wallet = selectedWallet else { return [] }
-        let ethAccount = DomainRegistry.accountRepository.find(
-            id: AccountID(tokenID: Token.Ether.id, walletID: wallet.id))!
+        let ethAccount = DomainRegistry.accountRepository
+            .find(id: AccountID(tokenID: Token.Ether.id, walletID: wallet.id))!
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift:398]: 😆 👍

**Review by DmitryBespalov — CHANGES_REQUESTED** [review]
changes needed

File: SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift (hunk 1 above)
Comment (sche) [reply SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift:30]: ok

File: SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift (hunk 2 above)
Comment (sche) [reply SafeUIKit/SafeUIKit/Components/InfoLabel/InfoLabel.swift]: ok

File: SafeUIKit/SafeAppUI/ReusableDetachedControllers/PaymentMethod/PaymentMethodHeaderView.swift (hunk 3 above)
Comment (sche) [reply SafeUIKit/SafeAppUI/ReusableDetachedControllers/PaymentMethod/PaymentMethodHeaderView.swift]: ok

File: MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift (hunk 4 above)
Comment (sche) [reply MultisigWallet/MultisigWalletApplication/WalletApplicationService.swift]: ok

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**